Lib/glyphsLib/builder/sources.py

```diff
@@ -80,6 +83,59 @@ def _to_designspace_source(self, master, is_regular):
     source.location = location
 
 
+def _to_designspace_source_layer(self):
+    brace_layers = collections.defaultdict(list)
+    layer_to_glyph = collections.defaultdict(list)
+    for glyph in self.font.glyphs:
+        for layer in glyph.layers:
+            if (
+                "{" in layer.name
+                and "}" in layer.name
+                and ".background" not in layer.name
+            ):
+                associated_master = self._sources[layer.associatedMasterId]
+                brace_layers[layer].append(associated_master)
+                layer_to_glyph[layer.name].append(glyph.name)
+
+    designspace = self._designspace
+    for brace_layer, masters in brace_layers.items():
+        brace_layer_name = brace_layer.name
+        layer_coordinates = [
+            int(c)
+            for c in brace_layer_name[
+                brace_layer_name.index("{") + 1 : brace_layer_name.index("}")
+            ].split(",")
+        ]
+        if len(layer_coordinates) != len(designspace.axes):
+            logger.warning(
+                "Glyph(s) %s, brace layer '%s' defines more or less locations than "
+                "there are design axes.",
```

Comment:
@madig I wonder if we should skip thus invalid brace layers instead?

Reply:
actually, I checked WorkSans and it has several brace layers specifying more values than there are axes... Anyway, better keep it like that: i.e. read in as many layer coords as there are axes, warn if their number they exceed the latter's; if less than that, maybe don't warn but simply imply the default value for the axis that was omitted?

Reply:
Ok. What happens if there are more values than axes?